fixtures/entrypoints-rpc-tests/tests/entrypoints.spec.ts

```diff
@@ -0,0 +1,825 @@
+import assert from "node:assert";
+import fs, { mkdir, writeFile } from "node:fs/promises";
+import os from "node:os";
+import path from "node:path";
+import dedent from "ts-dedent";
+import { Agent, fetch, setGlobalDispatcher } from "undici";
+import { test as baseTest, expect, vi } from "vitest";
+import { unstable_startWorkerRegistryServer } from "wrangler";
+import {
+	runWranglerDev,
+	runWranglerPagesDev,
+} from "../../shared/src/run-wrangler-long-lived";
+
+const timeoutAgent = new Agent({
+	connectTimeout: 500,
+	bodyTimeout: 500,
+	headersTimeout: 500,
+});
+setGlobalDispatcher(timeoutAgent);
+
+const tmpPathBase = path.join(os.tmpdir(), "wrangler-tests");
+
+type WranglerDevSession = Awaited<ReturnType<typeof runWranglerDev>>;
+type StartDevSession = (
+	files: Record<string, string>,
+	flags?: string[],
+	pagesPublicPath?: string
+) => Promise<{ url: URL; session: WranglerDevSession }>;
+
+export async function seed(root: string, files: Record<string, string>) {
+	for (const [name, contents] of Object.entries(files)) {
+		const filePath = path.resolve(root, name);
+		await mkdir(path.dirname(filePath), { recursive: true });
+		await writeFile(filePath, contents);
+	}
+}
+
+function waitFor<T>(callback: Parameters<typeof vi.waitFor<T>>[0]) {
+	// The default timeout of `vi.waitFor()` is only 1s, which is a little
+	// short for some of these tests, especially on Windows.
+	return vi.waitFor(callback, { timeout: 5_000, interval: 250 });
+}
+
+const test = baseTest.extend<{
+	tmpPath: string;
+	isolatedDevRegistryPort: number;
+	dev: StartDevSession;
+}>({
+	// Fixture for creating a temporary directory
+	async tmpPath({}, use) {
+		const tmpPath = await fs.realpath(await fs.mkdtemp(tmpPathBase));
+		await use(tmpPath);
+		await fs.rm(tmpPath, { recursive: true, maxRetries: 10 });
+	},
+	// Fixture for starting an isolated dev registry server on a random port
+	async isolatedDevRegistryPort({}, use) {
+		// Start a standalone dev registry server for each test
+		const result = await unstable_startWorkerRegistryServer(0);
+		const address = result.server.address();
+		assert(typeof address === "object" && address !== null);
+		await use(address.port);
+		await result.terminator.terminate();
+	},
+	// Fixture for starting a worker in a temporary directory, using the test's
+	// isolated dev registry
+	async dev({ tmpPath, isolatedDevRegistryPort }, use) {
```

Comment:
This doesn't seem to match the signature of StartDevSession

Am I missing something?